pkg/search/proxy/framework/runtime/framework.go

```diff
@@ -0,0 +1,52 @@
+package runtime
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"sort"
+
+	"k8s.io/apiserver/pkg/endpoints/request"
+
+	"github.com/karmada-io/karmada/pkg/search/proxy/framework"
+)
+
+// frameworkImpl select appropriate plugin to do `Connect()`
+type frameworkImpl struct {
+	plugins []framework.Plugin
+}
+
+// frameworkImpl is actually a Proxy
+var _ framework.Proxy = (*frameworkImpl)(nil)
+
+// NewFramework create instance of framework.Proxy with determined order of Plugin.
+func NewFramework(plugins []framework.Plugin) framework.Proxy {
+	sort.Slice(plugins, func(i, j int) bool {
+		return plugins[i].Order() < plugins[j].Order()
+	})
+
+	return &frameworkImpl{
+		plugins: plugins,
+	}
+}
+
+// Connect implements Proxy
+func (c *frameworkImpl) Connect(ctx context.Context, param framework.ConnectParam) (http.Handler, error) {
+	plugin, err := c.selectPlugin(param.RequestInfo)
+	if err != nil {
+		return nil, err
+	}
+
+	return plugin.Connect(ctx, param)
+}
+
+// selectPlugin return an appropriate Plugin by query Plugin.SupportRequest in order.
+func (c *frameworkImpl) selectPlugin(requestInfo *request.RequestInfo) (framework.Plugin, error) {
+	for _, plugin := range c.plugins {
+		if plugin.SupportRequest(requestInfo) {
+			return plugin, nil
+		}
+	}
+
+	return nil, errors.New("no plugin found for request")
```

Comment:
fmt.Errorf("no plugin found for request %v %v", method, gvr)

Reply:
```
fmt.Errorf("no plugin found for request: %v %v", requestInfo.Verb, requestInfo.Path)
```